core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -143,18 +143,37 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
     } else false
   }
 
-  private def aclMatch(operations: Operation, resource: Resource, principal: KafkaPrincipal, host: String, permissionType: PermissionType, acls: Set[Acl]): Boolean = {
+  private def aclMatch(operation: Operation, resource: Resource, principal: KafkaPrincipal, host: String, permissionType: PermissionType, acls: Set[Acl]): Boolean = {
     acls.find { acl =>
       acl.permissionType == permissionType &&
-        (acl.principal == principal || acl.principal == Acl.WildCardPrincipal) &&
-        (operations == acl.operation || acl.operation == All) &&
+        matchPrincipal(acl.principal, principal) &&
+        (operation == acl.operation || acl.operation == All) &&
         (acl.host == host || acl.host == Acl.WildCardHost)
     }.exists { acl =>
-      authorizerLogger.debug(s"operation = $operations on resource = $resource from host = $host is $permissionType based on acl = $acl")
+      authorizerLogger.debug(s"operation = $operation on resource = $resource from host = $host is $permissionType based on acl = $acl")
       true
     }
   }
 
+  /**
+    * @param valueInAcl KafkaPrincipal value present in Acl.
+    * @param input KafkaPrincipal present in the request.
+    * @return true if there is a match (including wildcard-suffix matching).
+    */
+  def matchPrincipal(valueInAcl: KafkaPrincipal, input: KafkaPrincipal): Boolean = {
+    (valueInAcl.getPrincipalType == input.getPrincipalType || valueInAcl.getPrincipalType == Acl.WildCardString) &&
+      (valueInAcl.getName.equals(input.getName) || valueInAcl.getName.equals(Acl.WildCardString)) // TODO
+  }
+
+  /**
+    * @param valueInZk Resource value stored on ZK.
+    * @param input Resource present in the request.
+    * @return true if there is a match (including wildcard-suffix matching).
+    */
+  def matchResource(valueInZk: Resource, input: Resource): Boolean = {
+    valueInZk.resourceType == input.resourceType && SecurityUtils.matchWildcardSuffixedString(valueInZk.name, input.name)
```

Comment:
Need to only call the wildcard-suffix matching logic if the valueInZk is wildcard-suffix